File: 2-pattrons-de-structures/bridge/code-java/implementations/concretes_implementations/Tv.java
package implementations.concretes_implementations;

import implementations.Device;

public class Tv implements Device {
    private boolean on = false;
    private int volume = 30;
    private int channel = 1;

    public boolean isEnabled() {
        return on;
    }

    public void enable() {
        on = true;
    }

    public void disable() {
        on = false;
    }

    public int getVolume() {
        return volume;
    }

    public void setVolume(int percent) {
        if (percent > 100) {
            this.volume = 100;
        } else {
            this.volume = percent;
        }
    }

    public int getChannel() {
        return channel;
    }

    public void setChannel(int channel) {
        this.channel = channel;
    }
}

File: 2-pattrons-de-structures/bridge/code-typescript/implementations/concretesImplementations/tv.ts
import { Device } from "../interface.js";

export class Tv implements Device {
  private isTvEnabled: boolean;
  private volume: number = 30;
  private channel: number = 1;

  constructor(isTvEnabled: boolean = false) {
    this.isTvEnabled = isTvEnabled;
  }

  isEnabled(): boolean {
    return this.isTvEnabled;
  }

  enable(): void {
    this.isTvEnabled = true;
  }

  disable(): void {
    this.isTvEnabled = false;
  }

  getVolume(): number {
    return this.volume;
  }

  setVolume(percent: number): void {
    this.volume = percent;
  }

  getChannel(): number {
    return this.channel;
  }

  setChannel(channel: number): void {
    this.channel = channel;
  }
}
